### PR TITLE
Add instructions on how to enable tracing in mbed OS 5

### PR DESCRIPTION
**Pull request by janjongboom**
@tommikas

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +133,36 @@ int main(void){
...
 
+## Enabling traces in mbed OS 5
+
+To enable traces in mbed OS 5, add the following to your ``mbed_app.json``:
+
+```json
+{
+   "target_overrides": {
+      "*": {
+         "mbed-trace.enable": 1
```
Comment (tommikas) [inline README.md]: The library is inside FEATURE_COMMON_PAL, so shouldn't that have to be added here as well?

File: README.md [hunk 2]
```diff
@@ -133,6 +133,36 @@ int main(void){
...
+      }
+   }
+}
+```
+
+Then instantiate the tracing module, and forward data to stdout by adding the following to your ``main.cpp``:
```
Comment (tommikas) [inline README.md]: Usage is covered in the "Usage example" section above.

File: README.md [hunk 3]
```diff
@@ -133,6 +133,36 @@ int main(void){
...
+
+```cpp
+#include "mbed-trace/mbed_trace.h"
+
+void trace_printer(const char* str) {
+   printf("%s\r\n", str);
+}
```
Comment (tommikas) [inline README.md]: Except for this bit...

**Comment by tommikas** [comment]
@jupe Could you have a look at this too please.

File: README.md (hunk 1 above)
Comment (tommikas) [inline README.md]: Checked to be sure, and it does need to be added. (I've seen some things compile without explicitly adding the relevant feature into the build, like network-socket API using code without including FEATURE_IPV4)

Anyway, `"target.features_add": ["COMMON_PAL"],` is missing here.

File: README.md (hunk 3 above)
Comment (tommikas) [inline README.md]: Providing the trace_printer function is also not strictly speaking necessary if all you want to do is get traces. It's currently used in a lot of apps to make cli prints and traces to play along better.

File: README.md (hunk 1 above)
Comment (jupe) [inline README.md]: yes, `COMMON_PAL` feature is missing. this might change in future, but at the moment it is required.

**Comment by janjongboom** [comment]
@tommikas @jupe In all fairness, I don't care where it goes as long as it's updated :p

**Comment by tommikas** [comment]
I figured as much :P Rather than try to specify exactly what changes you should do to have this merged in, I figured it'd be faster for me to make a quick PR with my own suggestion. I hope you don't mind. 

Could you take a look at https://github.com/ARMmbed/mbed-trace/pull/50 and let me know what you think?

**Comment by tommikas** [comment]
Merged in the other PR so I'll close this one.

**Pull request closed without merging**